Tasks/Common/sonarqube-common/sonarqube-common.ts

```diff
@@ -87,6 +90,19 @@ export function getSonarQubeEndpointFromInput(inputFieldName): SonarQubeEndpoint
     return new SonarQubeEndpoint(hostUrl, hostUsername, hostPassword);
 }
 
+// Returns, as an object, the contents of the 'report-task.txt' file created by SonarQube plugins
+// The returned object contains the following properties:
+//   projectKey, serverUrl, dashboardUrl, ceTaskId, ceTaskUrl
+export function getSonarQubeTaskReport(sonarPluginFolder: string): Map<string, string> {
```

Comment:
Thinking about it a bit more, I think a data class would be much better. Then we can keep the exact names of the key/value pairs that correspond to its fields in the data class rather than in the method that creates the build summary.

Accept, in progress.